### PR TITLE
Aine Longcoats

### PR DESCRIPTION
**Pull request by foxcumberland**
# Description

Double-breasted **longcoats** for almost everyone! Includes one for each unit of Command, three for Security Staff (Corpsman, Warden, Secoff), and four general-purpose ones (Brown, Black, Nanotrasen and an Arashi Labs 'medical' colored one, and of course a green version of the Captain's for Central Command.

Command, Corpsman and Warden Longcoats are added to their respective lockers. The basic Armored Longcoat has two copies in the SecTech between the Durathread Vest and Plate Carrier (as it is statistically analagous to them) while the four general longcoats are in the ClothesMate.

Designs are in media.

---

# TODO

Add at least the general longcoats to loadouts.

_Possibly_ add Corporate Jacket versions, need to discuss item bloat at some point as adding _a dozen_ jackets into the ClothesMate is going to make that list very, very long...

---

<details><summary><h1>Media</h1></summary>
<p>

Poster

</p>
</details>

---

# Changelog

:cl:

- Adds Longcoats
4 Common ones found in the clothesmate, Brown, Black, Nanotrasen, and an Arashi Labs 'medical cyan' one.
4 Armored Sec ones, with Secoff Armored coats available in the SecTech and unique ones found in HoS, Brigmedic and Warden lockers, Detective vendor has a copy of the Armored longcoat as well as an unarmored Brown longcoat
7 Command-colored ones to match their respective Cloaks: Captain, HoP, LO, RD, CMO, CE all have theirs available in their locker, and a Central Command one for admins.

### CODE REVIEW

**Comment by FoxxoTrystan** [comment]
Before i do any changes, this PR feels its will fit better in upsteam: https://github.com/Simple-Station/Einstein-Engines/

I have run also the test and RSI seme angry, this will need fixes.

**Comment by FoxxoTrystan** [comment]
And changlog will need to be different please.

**Comment by foxcumberland** [comment]
> Before i do any changes, this PR feels its will fit better in upsteam: https://github.com/Simple-Station/Einstein-Engines/
> 
> I have run also the test and RSI seme angry, this will need fixes.

I won't lie, I have no idea what any of that or the other review stuff means yet, I have no idea what a fail means and I don't know if that's somehow something I added causing a break or something else.

Changelog can be made real, I just had my brain in the wrong space as to what the changelog was referencing (updates to the pr rather than updates to the master)

Probably could be done upstream but I don't need the anxiety of trying to publish something to a group that I don't know, to people that I don't have any real contact with yet or am comfortable asking for help, when I'm just figuring this stuff out.

File: Resources/Prototypes/Catalog/Fills/Lockers/security.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@
       - id: ClothingShoesBootsJack
       - id: ClothingOuterCoatWarden
       - id: ClothingOuterWinterWarden
+      - id: ClothingLongcoatWarden
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Catalog/Fills/Lockers/security.yml:19]:
```suggestion
      - id: ClothingLongcoatWarden
```
```suggestion
      - id: ClothingLongcoatWarden # Floofstation
```

File: Resources/Prototypes/Catalog/Fills/Lockers/security.yml [hunk 2]
```diff
@@ -45,6 +46,7 @@
...
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Catalog/Fills/Lockers/security.yml:49]:
```suggestion
      - id: ClothingLongcoatWarden # Floofstation
```

File: Resources/Prototypes/Catalog/Fills/Lockers/security.yml [hunk 3]
```diff
@@ -119,6 +121,7 @@
       - id: ClothingUniformJumpskirtBrigmedic
 ##      - id: ClothingUniformJumpskirtOfLife # DeltaV - nah
 ##        prob: 0.1
+      - id: ClothingLongcoatBrigmedic
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Catalog/Fills/Lockers/security.yml:124]:
```suggestion
      - id: ClothingLongcoatBrigmedic # Floofstation
```

File: Resources/Prototypes/Catalog/Fills/Lockers/security.yml [hunk 4]
```diff
@@ -154,6 +157,7 @@
       - id: ClothingOuterHardsuitCombatCorpsman
       - id: ClothingUniformJumpsuitBrigmedic
       - id: ClothingUniformJumpskirtBrigmedic
+      - id: ClothingLongcoatBrigmedic
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Catalog/Fills/Lockers/security.yml:160]:
```suggestion
      - id: ClothingLongcoatBrigmedic # Floofstation
```

File: Resources/Prototypes/Catalog/VendingMachines/Inventories/clothesmate.yml [hunk 5]
```diff
@@ -95,6 +95,10 @@
     ClothingOuterFaCorporateJacket: 2 # Einstein Engines - Clothing addition
     ClothingOuterDdCorporateJacket: 2 # Einstein Engines - Clothing addition
     ClothingOuterBcCorporateJacket: 2 # Einstein Engines - Clothing addition
+    ClothingLongcoatNT: 2
+    ClothingLongcoatAL: 2
+    ClothingLongcoatBrown: 2
+    ClothingLongcoatBlack: 2
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Catalog/VendingMachines/Inventories/clothesmate.yml:101]:
```suggestion
    ClothingLongcoatNT: 2 # Floofstation
    ClothingLongcoatAL: 2 # Floofstation
    ClothingLongcoatBrown: 2 # Floofstation
    ClothingLongcoatBlack: 2 # Floofstation
```

File: Resources/Prototypes/Catalog/VendingMachines/Inventories/detdrobe.yml [hunk 6]
```diff
@@ -9,8 +9,10 @@
     ClothingUniformJumpsuitDetectiveGrey: 2
     ClothingUniformJumpskirtDetectiveGrey: 2
     ClothingOuterVestDetective: 2
+    ClothingLongcoatBrown: 1
+    ClothingLongcoatSec: 1
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Catalog/VendingMachines/Inventories/detdrobe.yml:13]:
```suggestion
    ClothingLongcoatBrown: 1 # Floofstation
    ClothingLongcoatSec: 1 # Floofstation
```

File: Resources/Prototypes/Catalog/VendingMachines/Inventories/sec.yml [hunk 7]
```diff
@@ -22,6 +22,7 @@
     ClothingHeadCage: 2 # Nyanotrasen - Insulative headgear
     ClothingOuterArmorPlateCarrier: 2 # DeltaV - moved body armour from SecDrobe to SecTech
     ClothingOuterArmorDuraVest: 2 
+    ClothingLongcoatSec: 2
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Catalog/VendingMachines/Inventories/sec.yml:25]:
```suggestion
    ClothingLongcoatSec: 2 # Floofstation
```

File: Resources/Prototypes/Floof/Entities/Clothing/OuterClothing/wintercoats.yml
Comment (FoxxoTrystan) [inline Resources/Prototypes/Floof/Entities/Clothing/OuterClothing/wintercoats.yml:1]: Are the armor here normal? as if i remember we avoid to add armor mostly to civillians

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]
Armor Stats need talk.
Copyright need to be set and not "modified later".
Comments are needed on Floof to not be overridden and so we know where its comes, follow suggestions.

File: Resources/Prototypes/Floof/Entities/Clothing/OuterClothing/wintercoats.yml
Comment (foxcumberland) [reply Resources/Prototypes/Floof/Entities/Clothing/OuterClothing/wintercoats.yml:1]: suggestions committed. For the armor values, I figured it'd be cute to have, but if it's a problem, I don't mind removing those values entirely.

**Comment by foxcumberland** [comment]
> I do wana note that upsteam is not wizden but upsteam.

I know it's not wizden, but otherwise this... just confused me. Is upstream not what we call the EE code/team, because That's what I was referring to when I meant people I don't know.

**Comment by FoxxoTrystan** [comment]
RSI / Linter test failed, click details and fix the issues displayed.
If unble to find the error, feel free to ping me on discord.

**Comment by Fansana** [comment]
I really like this. We can merge this upstream if need be that shouldn't be a problem.
Please fix the following errors so we can merge it
Resources/Textures/Floof/Clothing/OuterClothing/WinterCoats/longcoatAL.rsi: PNG not defined in metadata: Icon.png
Resources/Textures/Floof/Clothing/OuterClothing/WinterCoats/longcoatAL.rsi: icon: failed to open state icon.png

**Review by Fansana — APPROVED** [review]

**Comment by FoxxoTrystan** [comment]
RSI:
Resources/Textures/Floof/Clothing/OuterClothing/WinterCoats/longcoatAL.rsi: PNG not defined in metadata: Icon.png
Resources/Textures/Floof/Clothing/OuterClothing/WinterCoats/longcoatAL.rsi: icon: failed to open state icon.png

YAML Linter:
Unable to load RSI '/Textures/Floof/Clothing/OuterClothing/WinterCoats/longcoatAL.rsi'.

Check failed tests for more information.

**Comment by foxcumberland** [comment]
RSI/YAML fixed, I believe. Got to love all that angry error text just for a simple filename capitalization goof.

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]
Linter still uhappy:

Error: /Prototypes/Catalog/VendingMachines/Inventories/clothesmate.yml(98,5)  PrototypeID ClothingLongcoatNT for type Robust.Shared.Prototypes.EntityPrototype at Line: 98, Col: 5 not found
Error: /Prototypes/Catalog/VendingMachines/Inventories/clothesmate.yml(99,5)  PrototypeID ClothingLongcoatAL for type Robust.Shared.Prototypes.EntityPrototype at Line: 99, Col: 5 not foun

**Comment by FoxxoTrystan** [comment]
> RSI/YAML fixed, I believe. Got to love all that angry error text just for a simple filename capitalization goof.
> 
> Edit: Nevermiiind it was showing me green until it failed...

Yeah because its your first PR here, i have to aprove the tests myself.

**Comment by foxcumberland** [comment]
Should be good now, LongcoatNT and LongcoatAL were newer names before that the files got but the prototype didn't.

**Review by FoxxoTrystan — APPROVED** [review]

**Pull request merged**